.changeset/khaki-books-rescue.md

```diff
@@ -0,0 +1,33 @@
+---
+'@cloudflare/next-on-pages': minor
+---
+
+add new `utils` sum-module with `getRequestCfProperties` and `getRequestExecutionContext`
+
+add a new `utils` module that will provide utilities that will allow developer to utilize
+better next-on-pages and the cloudflare platform.
+
+alongside the `utils` two functions have been introduced `getRequestCfProperties` and
+`getRequestExecutionContext`, the former provides to developers the `cf` object from
+the request, whilst the latter the execution context object (`ctx`), both results
+are properly typed (with the types coming from `@cloudflare/workers-types`).
+
+Note that naturally both only work on the server, they throw an exception if run on the client,
+where there is not request to get the objects from.
+
+Usage example:
+
+```ts
+import {
+	getRequestCfProperties,
+	getRequestExecutionContext,
+} from '@cloudflare/next-on-pages/utils';
+
+// ...
+
+const cf = getRequestCfProperties();
+// access cf.colo, cf.city, etc...
+
+const ctx = getRequestExecutionContext();
+// ctx.waitUntil(myPromise), etc...
```

Comment:
We should show this in a useful context (e.g. being used in an App Router page render).

Reply:
This is just the changeset, I will look into docs later if that's ok 🙂 

(or do you feel strongly that the changeset example should be good?)